### PR TITLE
Move existing "Prefer functional components over classes" guide

### PR DESCRIPTION
**Pull request by pavelvlasov**

### CODE REVIEW

File: guides/architecture/general/components/prefer-functional-components-over-classes.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Prefer functional components over classes
+
+React 16.8 is bringing hooks, so there are some strong reasons to consider the usage of functional components over classes.
+
+## Why not to use it
```
Comment (nadiam84) [inline guides/architecture/general/components/prefer-functional-components-over-classes.md]: "Why not to use classes" I assume?

File: guides/architecture/general/components/prefer-functional-components-over-classes.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+exports.default = _default;
+```
+
+You can check it out in [babel playground](https://babeljs.io/en/repl)
+
+**When to use it**
+
+Migration to hooks might take a while, so it makes sense to keep using classes for statefull components or when lifecycle methods are required.
+
+**Lambdas vs Functions**
```
Comment (nadiam84) [inline guides/architecture/general/components/prefer-functional-components-over-classes.md]: Can it be reworded into something like `prefer functions over lambdas`, same as `components over classes`?

File: guides/architecture/general/components/prefer-functional-components-over-classes.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+MyComponent.defaultProps = {
+  foo: 'bar'
+};
+var _default = MyComponent;
+exports.default = _default;
+```
+
+You can check it out in [babel playground](https://babeljs.io/en/repl)
+
+**When to use it**
```
Comment (nadiam84) [inline guides/architecture/general/components/prefer-functional-components-over-classes.md]: When to use classes :)

File: guides/architecture/general/components/prefer-functional-components-over-classes.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+React 16.8 is bringing hooks, so there are some strong reasons to consider the usage of functional components over classes.
+
+## Why not to use it
+
+**Bundle size**.
+The main reason for the change is the fact, that classes as components significantly increase bundle size.
+
+If we take the following class:
+```js
+import React, {Component} from 'react';
```
Comment (nadiam84) [inline guides/architecture/general/components/prefer-functional-components-over-classes.md]: `{ Component }` ;p

**Comment by nadiam84** [comment]
@pavelvlasov can you also add link to this guide here: https://github.com/atlassian/tangerine/tree/master/guides

File: guides/architecture/general/components/prefer-functional-components-over-classes.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
+# Prefer functional components over classes
+
+React 16.8 is bringing hooks, so there are some strong reasons to consider the usage of functional components over classes.
```
Comment (nadiam84) [inline guides/architecture/general/components/prefer-functional-components-over-classes.md]: Hooks are not the only reason though, right? Smaller bundle sizes and less boilerplate is another one. It's just with hooks we're able to utilise functional components to the fullest

File: guides/architecture/general/components/prefer-functional-components-over-classes.md (hunk 1 above)
Comment (pavelvlasov) [reply guides/architecture/general/components/prefer-functional-components-over-classes.md]: Good call, I actually copied it over from https://github.com/atlassian/tangerine/blob/master/guides/architecture/general/components/prefer-component-over-pure-component.md

File: guides/architecture/general/components/prefer-functional-components-over-classes.md (hunk 5 above)
Comment (pavelvlasov) [reply guides/architecture/general/components/prefer-functional-components-over-classes.md]: 👍good call! rephrased it

**Pull request merged**